output/transform/postgres_plans.go

```diff
@@ -0,0 +1,107 @@
+package transform
+
+import (
+	"time"
+
+	snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func transformPostgresPlans(s snapshot.FullSnapshot, newState state.PersistedState, diffState state.DiffState, transientState state.TransientState, roleOidToIdx OidToIdx, databaseOidToIdx OidToIdx, statementKeyToIdx StatementKeyToIdx) snapshot.FullSnapshot {
+	planInformations := []*snapshot.QueryPlanInformation{}
+	planStats := []*snapshot.QueryPlanStatistic{}
+	for pKey, plan := range transientState.Plans {
+		sKey := postgresStatementKey{
+			databaseOid: pKey.DatabaseOid,
+			userOid:     pKey.UserOid,
+			queryID:     pKey.QueryID,
+		}
+		queryIdx, exist := statementKeyToIdx[sKey]
```

Comment:
As discussed, what can happen here is that we have multiple statement keys pointing to the same queryIdx (either because of a case where top_level is the only difference, or because we have multiple query IDs pointing to the same fingerprint, and as such queryIdx).

The easiest solution might be to make both `planInformations` and `planStats` a map that's indexed by `queryIdx` and `planId` (or a `planIdx` if you introduce that), and then do the addition of stats similar to how `groupStatements` does it.

Reply:
For the plan information we could keep the more recently captured plan (if we have two plans for the same query index and plan ID).